### PR TITLE
CGP-2 Support EPICS area under Git

### PR DESCRIPTION
**Pull request by mfurseman**
We need the ADE to be able to release modules into the epics area.

### CODE REVIEW

**Comment by mfurseman** [comment]
I don't think I took enough care over your requirement. I've pushed a commit that should allow building on top of an existing module directory. Contrary to other modules, the epics release script now creates a `$_module` directory and checks `$_version` out in it, as opposed to cloning into a directory called `$_module/$_version`.

**Comment by ulrikpedersen** [comment]
Yes, that looks more appropriate in the epics.sh script. It's all a bit convoluted, so please test it somewhere (not in prod) with a few different R3.14.12.7dlsNN tags in there to make sure it does the right thing...

**Comment by mfurseman** [comment]
I tested these changes building into /scratch. The releases stack as expected, updating the git repo then calling `make && make clean && make install`.

**Comment by ulrikpedersen** [comment]
OK, try a test-release on the RHEL7 build server next. Using -r 7 -tT

**Comment by mfurseman** [comment]
Seems to have built correctly into
/dls_sw/work/etc/build/test/build_20180208-100139_uxj42447_epics_base_3-14-12-7/R3.14.12.7/base

File: dls_ade/dlsbuild.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 build_servers = {
     "Linux": {
         "redhat6-x86_64": ["R3.14.12.3"],
-        "redhat7-x86_64": ["R3.14.12.6"]
+        "redhat7-x86_64": ["R3.14.12.7"]
```
Comment (ulrikpedersen) [inline dls_ade/dlsbuild.py]: Its probably best to pick the latest like you do here: R3.14.12.7.
However, we then also have to remember to modify our global profile (*only* during shutdown!) as I added EPICS support for RHEL7 back when R3.14.12.6 was the latest... So when you log into a RHEL7 machine you get an environment which expects R3.14.12.6:

```
[up45@pc0002 ~]$ env | grep -i epics
EPICS_MBA_TEMPLATE_TOP=/dls_sw/epics/R3.14.12.6/base/templates/makeBaseApp/top
EPICS_BASE=/dls_sw/epics/R3.14.12.6/base
EPICS_EXTENSIONS=/dls_sw/epics/R3.14.12.6/extensions
PATH=/home/up45/bin:/home/up45/bin/scripts:/dls_sw/epics/R3.14.12.6/base/bin/linux-x86_64:/dls_sw/epics/R3.14.12.6/extensions/bin/linux-x86_64:/dls_sw/prod/tools/RHEL7-x86_64/defaults/bin:/usr/lib64/qt-3.3/bin:/home/up45/perl5/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/var/cfengine/bin
EPICS_HOST_ARCH=linux-x86_64
_LMFILES_=/usr/share/Modules/modulefiles/use.own:/dls_sw/etc/modulefiles/controls-tools:/dls_sw/etc/modulefiles/controls:/dls_sw/etc/modulefiles/epics/3.14.12.6:/dls_sw/etc/modulefiles/controls_dev
LOADEDMODULES=use.own:controls-tools:controls:epics/3.14.12.6:controls_dev
DLS_EPICS_RELEASE=R3.14.12.6
EPICS_CA_MAX_ARRAY_BYTES=6000000
```

Sorry.

**Pull request merged**